lib/csf-tools/src/CsfFile.ts

```diff
@@ -388,8 +388,18 @@ export class CsfFile {
     });
 
     if (self._namedExportsOrder) {
+      const unsortedExports = Object.keys(self._storyExports);
       self._storyExports = sortExports(self._storyExports, self._namedExportsOrder);
       self._stories = sortExports(self._stories, self._namedExportsOrder);
+
+      const sortedExports = Object.keys(self._storyExports);
+      if (unsortedExports.length !== sortedExports.length) {
```

Comment:
Will this take into account the following scenario:

> User sets the same amount of stories as they have as keys in `__namedExportsOrder`, but they have different ids:

```js
export const Primary = {}
export const Secondary = {}

export const __namedExportsOrder = [
  'Default',
  'Secondary'
]
```

If not, something like this could be more precise?

```js
Object.keys(self._storyExports).every(exp => self._namedExportsOrder.includes(exp))
```

Reply:
Yes. The "sorting" logic above turns into a filter if any keys are missing, so count is good enough to catch any problems. And fast.